linera-chain/src/chain.rs

```diff
@@ -375,10 +375,10 @@ where
         &mut self,
         local_time: Timestamp,
         query: Query,
-        incoming_execution_requests: &mut futures::channel::mpsc::UnboundedReceiver<
-            ExecutionRequest,
+        incoming_execution_requests: Option<
+            &mut futures::channel::mpsc::UnboundedReceiver<ExecutionRequest>,
         >,
-        runtime_request_sender: &mut std::sync::mpsc::Sender<ServiceRuntimeRequest>,
+        runtime_request_sender: Option<&mut std::sync::mpsc::Sender<ServiceRuntimeRequest>>,
```

Comment:
Optional: this feels like a code smell. Could we pack these two fields into a new `struct ServiceRuntimeChannel`  or `struct ServiceRuntimeEndpoint`?

Reply:
Yes, this makes sense